src/main/java/com/whiletrue/tododemo/controller/TaskController.java

```diff
@@ -0,0 +1,57 @@
+package com.whiletrue.tododemo.controller;
+
+import com.whiletrue.tododemo.dto.Task;
+import com.whiletrue.tododemo.dto.TaskResponse;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/tasks")
+public class TaskController {
+
+    private static Task hardcodedTask = new Task("test", "test description", LocalDateTime.now().plusDays(2), "system", "elena", true);
+
+    @PostMapping
+    public TaskResponse createTask(@RequestBody Task task) {
+        // TODO create logic for inserting into DB
+        TaskResponse response = generateTaskResponse(task);
+        return response;
+    }
+
+    @GetMapping
+    public List<TaskResponse> getTasks() {
+        // TODO create logic for retrieving tasks from DB
+        List<TaskResponse> tasks = new ArrayList<>();
+        tasks.add(generateTaskResponse(hardcodedTask));
+        tasks.add(generateTaskResponse(hardcodedTask));
+        return tasks;
+    }
+
+    @GetMapping("/{taskId}")
+    public TaskResponse getTask(@PathVariable("taskId") Long taskId) {
+        // TODO create logic for retrieving task from DB
+        TaskResponse updatedTask = generateTaskResponse(hardcodedTask);
+        return updatedTask;
+    }
+
+    @PutMapping("/{taskId}")
+    public TaskResponse updateTask(@PathVariable("taskId") Long taskId,
+                           @RequestBody Task task) {
+        // TODO create logic for updating in DB
+        TaskResponse updatedTask = generateTaskResponse(task);
+        return updatedTask;
+    }
+
+    @DeleteMapping("/{taskId}")
+    public void deleteTask(@PathVariable("taskId") Long taskId) {
+        // TODO create logic for deleting from DB
+    }
+
+    private TaskResponse generateTaskResponse(Task task) {
```

Comment:
Use constructor in TaskResponse for mapping

Reply:
fixed